src/exporters/s5.py

```diff
@@ -0,0 +1,492 @@
+        """
+        assert (
+            variable in self.dataset_reference["variable"]
+        ), f"Variable: {variable} is not in the valid variables for this \
+            dataset. Valid variables: {self.dataset_reference['variable']}"
+
+        # setup the default selection request
+        processed_selection_request = {
+            "format": "grib",
+            "originating_centre": "ecmwf",
+            "system": "5",
+            "variable": [variable],
+        }
+
+        # add product_type if required
+        if self.product_type is not None:
+            processed_selection_request.update({"product_type": [self.product_type]})
+
+        init_times_dict = self.get_s5_initialisation_times(
+            self.granularity, min_year=min_year, max_year=max_year,
+            min_month=min_month, max_month=max_month
+        )
+
+        for key, val in init_times_dict.items():
+            processed_selection_request[key] = val
+
+        # get the leadtime information
+        leadtimes_dict = self.get_s5_leadtimes(max_leadtime)
+        for key, val in leadtimes_dict.items():
+            processed_selection_request[key] = val
+
+        # by default, we investigate Kenya
+        kenya_region = get_kenya()
+        processed_selection_request["area"] = self.create_area(kenya_region)
+
+        # TODO: do we want this level of flexibility for the user? why add complexity?
+        # # update with user arguments
+        # if selection_request is not None:
+        #     for key, val in selection_request.items():
+        #         if key in {'day', 'hour'}:
+        #             warnings.warn(f'Overriding default {key} values. '
+        #                           f'The ERA5Exporter assumes all days '
+        #                           f'and times (the default) are being downloaded')
+        #         val = self._check_iterable(val, key)
+        #         processed_selection_request[key] = [self._correct_input(x, key) for x in val]
+        if self.product_type is None:
+            keys = [k for k in processed_selection_request.keys()]
+            assert (
+                "product_type" not in keys
+            ), f"product type should not be in selection_request keys\
+            for dataset {self.dataset}"
+
+        return processed_selection_request
+
+    @staticmethod
+    def get_dataset(granularity: str, pressure_level: bool) -> str:
+        if granularity == "monthly":
+            return (
+                "seasonal-monthly-pressure-levels"
+                if pressure_level
+                else "seasonal-monthly-single-levels"
+            )
+        elif granularity == "hourly":
+            return (
+                "seasonal-original-pressure-levels"
+                if pressure_level
+                else "seasonal-original-single-levels"
+            )
+        else:
+            assert False, f"No dataset can be created with \
```

Comment:
It seems like `granularity` (which is defined at initialization) can only be `monthly` or `hourly`. Would it make sense to check that the string being passed at initialization is one of the two, instead of here?
This way, the check only happens once, straight away.

Reply:
great idea